src/Android/Xamarin.Android/Samples/Augmented Reality/NavigateAR/RouteViewerAR.cs

```diff
@@ -0,0 +1,293 @@
+
+            // Set up control references.
+            _arSceneView = FindViewById<ARSceneView>(ArcGISRuntime.Resource.Id.arView);
+            _helpLabel = FindViewById<TextView>(ArcGISRuntime.Resource.Id.helpLabel);
+            _calibrateButton = FindViewById<Button>(ArcGISRuntime.Resource.Id.calibrateButton);
+            _navigateButton = FindViewById<Button>(ArcGISRuntime.Resource.Id.navigateStartButton);
+            _calibrationView = FindViewById(ArcGISRuntime.Resource.Id.calibrationView);
+            _headingSlider = FindViewById<JoystickSeekBar>(ArcGISRuntime.Resource.Id.headingJoystick);
+            _altitudeSlider = FindViewById<JoystickSeekBar>(ArcGISRuntime.Resource.Id.altitudeJoystick);
+
+            // Create the custom location data source and configure the AR scene view to use it.
+            _locationDataSource = new MSLAdjustedARLocationDataSource(this);
+            _locationDataSource.AltitudeMode = MSLAdjustedARLocationDataSource.AltitudeAdjustmentMode.NmeaParsedMsl;
+            _arSceneView.LocationDataSource = _locationDataSource;
+
+            // Listen for location changes to enable route tracking.
+            _locationDataSource.LocationChanged += LocationDataSource_LocationChanged;
+
+            // Disable plane rendering and visualization.
+            _arSceneView.ArSceneView.PlaneRenderer.Enabled = false;
+            _arSceneView.ArSceneView.PlaneRenderer.Visible = false;
+
+            // Configure button click events.
+            _navigateButton.Click += (o, e) => StartTurnByTurn();
+            _calibrateButton.Click += (o, e) => IsCalibrating = !IsCalibrating;
+
+            // Configure calibration sliders.
+            _headingSlider.DeltaProgressChanged += HeadingSlider_DeltaProgressChanged;
+            _altitudeSlider.DeltaProgressChanged += AltitudeSlider_DeltaProgressChanged;
+        }
+
+        private void AltitudeSlider_DeltaProgressChanged(object sender, DeltaChangedEventArgs e)
+        {
+            _altitudeOffset += e.deltaProgress;
+            _locationDataSource.AltitudeOffset = _altitudeOffset;
+        }
+
+        private void HeadingSlider_DeltaProgressChanged(object sender, DeltaChangedEventArgs e)
+        {
+            Camera camera = _arSceneView.OriginCamera;
+
+            double heading = camera.Heading + e.deltaProgress;
+
+            Camera newCamera = camera.RotateTo(heading, camera.Pitch, camera.Roll);
+
+            _arSceneView.OriginCamera = newCamera;
+        }
+
+        private async void LocationDataSource_LocationChanged(object sender, Location e)
+        {
+            if (_routeTracker != null)
+            {
+                try
+                {
+                    // Pass location change to the route tracker.
+                    await _routeTracker.TrackLocationAsync(e);
+                }
+                catch (Exception ex)
+                {
+                    ShowMessage("Failed to update current location", "error");
```

Comment:
You can replace the ShowMessage method with
`new AlertDialog.Builder(this).SetMessage("Failed to update current location").SetTitle("Error").Show();`